src/docs/proposals/5496-batch-span-ingestion-with-events.mdx

```diff
@@ -0,0 +1,337 @@
+        ...
+    ]
+}
+```
+
+Transaction spans are distinguished from regular spans by **TODO**.
+
+The `Span` serialization does not change from the current format.
+
+Compared to the OpenTelemetry tracing protocol explained in the [introduction](#introduction) above, this proposal flattens instrumentation library and resource metadata into a single place, the top-level event fields. While SDKs that have multiple layers (for example, React Native events may come from and through different SDKs (JS, Java, C++, etc) depending on the layer where an event was created) could benefit from the extra level of indirection, it is out of scope of this proposal to deal with that.
+
+To support the proposed changes, two different implementation directions are proposed, each variant bringing along their own implications.
+
+Note that the variants are not necessarily mutually exclusive, both could co-exist. In fact, they can be seen as additive. A valid approach would be to start with Variant A, and later implement Variant B.
+
+### Variant A: Buffering in Relay
+
+This proposal implementation variant moves the downsides of buffering full transactions from SDKs into Relay.
+
+It is an increased operational risk as it directly affects Relay's memory consumption. The risk is mitigated by, for example, having customers run their own Relay instance that would operate pretty much like an OpenTelemetry Collector. It would be an ingestion point that is specific to an organization, and traffic to Sentry-controlled Relays would only accept full transactions, eliminating the need for buffering.
+
+The diagram below concisely explains the data ingestion flow of this variant implementation:
+
+```
+┌─────────────┐    ┌─────────────────────┐           xxxx   x          ┌──────────┐
+│             │    │                     │       xxxx         xxxxx    │          │
+│ Application │    │ User/Org-controlled │       x                x    │ Sentry   │
+│      +      ├───►│                     ├───►xx     Internet     x───►│          │
+│ Sentry SDK  │    │      Relay          │   xx                  xx    │ Relay(s) │
+│             │    │                     │     xxxxx           xxx     │          │
+└─────────────┘    └─────────────────────┘         xxxxxxxxxxxx        └──────────┘
+ Event types:
+ <error, transaction, spans>      <error, transaction>
+ ────────────────────────────►    ─────────────────────────────────────────►
+ ```
+
+As depicted, the `spans` event type would only exist/be accepted in the first Relay instance. Upstream Relays would not need to perform buffering.
+
+Note that this variant also works if Sentry accepts `spans` events in its front-facing Relay.
+
+Relay could decide to accept or reject span trees that have no transaction as the root. When accepting, it could choose to automatically promote any span tree root to a transaction, given that the data is already ingested and [billing concerns](#billing-impact) have been sorted out.
```

Comment:
@rhcarvalho Does the span tree promotion occur on the first Relay of the Relay chain (i.e. Relay to Relay)? This specific detail will matter on how we derive additional information based on the spans, and if we need to re-derive them again. Span operation breakdowns is an example of derived information.